bugsnag-plugin-android-ndk/src/main/java/com/bugsnag/android/NativeEventDecoder.kt

```diff
@@ -143,6 +147,69 @@ internal object NativeEventDecoder {
         event.setUser(id, email, name)
     }
 
+    private fun decodeError(eventBytes: ByteBuffer, event: Event) {
+        eventBytes.realign()
+        val errorClass = eventBytes.getCString(64)
+        val errorMessage = eventBytes.getCString(256)
+        val type = eventBytes.getCString(32)
+        val frameCount = eventBytes.getNativeSize()
+        val stacktrace = Array(BUGSNAG_FRAMES_MAX) { decodeFrame(eventBytes) }
+            .take(frameCount.toInt())
+
+        event.errors.map {
+            it.errorClass = errorClass
+            it.errorMessage = errorMessage
+            it.type = ErrorType.values()
+                .find { errorType -> errorType.name == type } ?: ErrorType.UNKNOWN
+            it.stacktrace.mapIndexed { index, frame ->
+                frame.method = stacktrace[index].method
+                frame.file = stacktrace[index].file
+                frame.lineNumber = stacktrace[index].lineNumber
+                frame.frameAddress = stacktrace[index].frameAddress
+                frame.symbolAddress = stacktrace[index].symbolAddress
+                frame.loadAddress = stacktrace[index].loadAddress
+                frame.codeIdentifier = stacktrace[index].codeIdentifier
+                if (frame.isPC == true) {
+                    frame.isPC = true
+                }
+            }
+        }
+    }
+
+    private fun decodeFrame(eventBytes: ByteBuffer): Stackframe {
+        eventBytes.realign()
+        val frameAddress = eventBytes.getNativeLong()
+        val symbolsAddress = eventBytes.getNativeLong()
+        val loadAddress = eventBytes.getNativeLong()
+        val lineNumber = eventBytes.getNativeLong()
+        val fileName = eventBytes.getCString(256)
+        val method = eventBytes.getCString(256)
+        val codeIdentifier = eventBytes.getCString(256)
+        val result = Stackframe(
+            method = null,
+            file = null,
+            lineNumber = lineNumber,
+            inProject = null,
+            code = null,
+            columnNumber = null
+        )
+        if (fileName.isNotEmpty()) {
+            result.file = fileName
+        }
+        if (method.isEmpty()) {
+            result.method = frameAddress.toString(20)
```

Comment:
This is not going to produce the correct address. See [JsonUtil](https://github.com/bugsnag/bugsnag-android/blob/4b946df726bf3242639eecacb1931574071f53b7/bugsnag-android-core/src/main/java/com/bugsnag/android/internal/JsonHelper.kt#L103-L111) for the correct implementation.